grakn-graph/src/main/java/ai/grakn/factory/AbstractInternalFactory.java

```diff
@@ -34,6 +34,8 @@
     protected G graph = null;
     private G batchLoadingGraph = null;
 
+    private Boolean lastGraphBuiltBatchLoading = null;
```

Comment:
Would it make sense for this to be `false` initially? It might make the logic easier.